src/wp-includes/block-template-utils.php

```diff
@@ -507,6 +507,64 @@ function _remove_theme_attribute_from_template_part_block( &$block ) {
 	}
 }
 
+/**
+ * Gets the contents for the given template file path, using a transient cache.
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param string $template_file_path Absolute path to a theme template file.
+ * @return string The template file contents.
+ */
+function _get_block_template_file_content( $template_file_path ) {
+	$theme = wp_get_theme();
+
+	$template_file_path = wp_normalize_path( $template_file_path );
+	$theme_dir          = wp_normalize_path( $theme->get_stylesheet_directory() );
+
+	if ( str_starts_with( $template_file_path, $theme_dir ) ) {
+		$relative_path = substr( $template_file_path, strlen( $theme_dir ) );
```

Comment:
Could use `str_replace()` here and below, similarly to how we're doing it in `register_core_block_style_handles()`:

```suggestion
		$relative_path = str_replace( $theme_dir, '', $template_file_path );
```

Reply:
For this purpose, `str_replace()` is less precise than `substr()`, since we want to replace the beginning of the string only, not the partial path wherever it is included in the string.

This is obviously an edge case consideration, but has been pointed out to me in the past, and it makes sense. We shouldn't use `str_replace()` if we want to only replace a single specific instance of the substring.

Reply:
Hmmm...even so, if `$theme_dir` is not at the beginning of the string, then the `substr()` function wouldn't remove what we intend to, which could be _more_ confusing. You already have this inside an if statement that assures that `$template_file_path` starts with `$theme_dir`, so it really seems like both approaches are the same, with no possibility for edge cases to occur.

Reply:
@joemcgill Let me clarify what I mean:
* `str_starts_with( '/dirname/dirname', '/dirname' )` results in `true`
* `substr( '/dirname/dirname', strlen( '/dirname' ) )` results in the expected `/dirname`
* `str_replace( '/dirname', '', '/dirname/dirname' )` results in `''`, which would be incorrect

Obviously this is not realistic to happen here since a file wouldn't be within a directory within the theme that has the same folder path as the overall theme directory. But if anything, it's safer to use `substr()` here over `str_replace()` to prevent it. IMO `str_replace()` is less precise since the intention here is to replace the instance of `$theme` that the string starts with, not _any_ instance of it within the string.

Anyway, this is a nit-pick discussion 😂